dash/development/_py_components_generation.py

```diff
@@ -91,18 +96,27 @@ def __init__(self, {default_argtext}):
     else:
         default_argtext = ""
         argtext = "**args"
-    default_argtext += ", ".join(
-        [
-            (
-                f"{p:s}=Component.REQUIRED"
-                if props[p]["required"]
-                else f"{p:s}=Component.UNDEFINED"
+    default_arglist = [
+        (
+            f"{p:s}=Component.REQUIRED"
+            if props[p]["required"]
+            else f"{p:s}=Component.UNDEFINED"
+        )
+        for p in prop_keys
+        if not p.endswith("-*") and p not in python_keywords and p != "setProps"
+    ]
+
+    if max_props:
+        final_max_props = max_props - (1 if "children" in props else 0)
+        if len(default_arglist) > final_max_props:
+            default_arglist = default_arglist[:final_max_props]
+            docstring += (
+                "\n\nNote: due to the large number of props for this component,"
+                "\nnot all of them appear in the constructor signature, but"
+                "\nthey may still be used as keyword arguments."
```

Comment:
The newline at the beginning is kinda weird, it would be more readable to put them at the end of the line.

Reply:
Good point - especially `\nnot` looks weird. I did this because I needed some newlines at the start of the string, but better to put those on their own and the others at the end of their lines -> https://github.com/plotly/dash/pull/2029/commits/b43044df12a4b4a10494e226f5145880c2be70be